api/src/main/java/ca/bc/gov/educ/api/graduation/model/report/ReportRequest.java

```diff
@@ -2,20 +2,30 @@
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonPropertyOrder;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.codehaus.jackson.annotate.JsonTypeInfo;
 
-@JsonPropertyOrder({ "options", "data"})
-public class ReportRequest {
-    private ReportOptions options;
-    private ReportData reportData;
+import java.io.Serializable;
 
-    @JsonProperty("options")
-    public ReportOptions getOptions() { return options; }
-    public void setOptions(ReportOptions value) { this.options = value; }
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonPropertyOrder({ "options", "data"})
+public class ReportRequest implements Serializable {
 
-    @JsonProperty("data")
-    public ReportData getData() { return reportData; }
-    public void setData(ReportData value) { this.reportData = value; }
-}
+	private ReportData data;
+	private ReportOptions options;
 
```

Comment:
Hi Sreepad, also make ReportData and ReportOptions implement Serializable for serialization safety. Thanks.